apps/financial-aid/web-veita/src/components/TableBody/TableBody.tsx

```diff
@@ -1,106 +1,103 @@
 import React from 'react'
-import { Text, Box, Button } from '@island.is/island-ui/core'
-import Link from 'next/link'
+import { Box, Text, Button } from '@island.is/island-ui/core'
 
 import * as tableStyles from '../../sharedStyles/Table.css'
 import cn from 'classnames'
-
-import {
-  Application,
-  getState,
-  getMonth,
-  ApplicationState,
-} from '@island.is/financial-aid/shared/lib'
-
-import {
-  GeneratedProfile,
-  GenerateName,
-} from '@island.is/financial-aid-web/veita/src/components'
-import {
-  calcDifferenceInDate,
-  getTagByState,
-} from '@island.is/financial-aid-web/veita/src/utils/formHelper'
+import { Colors } from '@island.is/island-ui/theme'
 
 interface PageProps {
-  application: Application
+  items: React.ReactNode[]
   index: number
-  onApplicationUpdate: (
-    applicationId: string,
-    state: ApplicationState,
-  ) => Promise<void>
+  identifier: string
+  onClick?: () => void
+  hasMaxWidth?: boolean
 }
 
-const TableBody = ({ application, index, onApplicationUpdate }: PageProps) => {
+type TextVariants =
+  | 'default'
+  | 'small'
+  | 'h1'
+  | 'h2'
+  | 'h3'
+  | 'h4'
+  | 'h5'
+  | 'intro'
+  | 'eyebrow'
+
+const TextTableItem = (
+  variant?: TextVariants,
+  text?: string | number,
+  color?: Colors,
+) => {
   return (
-    <Link href={'application/' + application.id}>
-      <tr
-        className={`${tableStyles.link} contentUp`}
-        style={{ animationDelay: 55 + 3.5 * index + 'ms' }}
+    text && (
+      <Text variant={variant} color={color}>
+        {text}
+      </Text>
+    )
+  )
+}
+
+const ActivationButtonTableItem = (
+  title: string,
+  loading: boolean,
+  onClick: () => void,
+  destructiveColorScheme: boolean,
+) => {
+  return (
+    <Box>
+      <Button
+        onClick={(event) => {
+          event.stopPropagation()
+          onClick()
+        }}
+        variant="text"
+        loading={loading}
+        colorScheme={destructiveColorScheme ? 'destructive' : 'light'}
       >
-        <td
-          className={cn({
-            [`${tableStyles.tablePadding} ${tableStyles.firstChildPadding}`]: true,
-          })}
-        >
-          <Box className={tableStyles.rowContent} alignItems="center">
-            <GeneratedProfile size={32} nationalId={application.nationalId} />
-            <Box marginLeft={2}>
-              <Text variant="h5">{GenerateName(application.nationalId)}</Text>
-            </Box>
-          </Box>
-        </td>
+        {title}
+      </Button>
+    </Box>
+  )
+}
 
+const TableBody = ({
+  items,
+  index,
+  identifier,
+  onClick,
+  hasMaxWidth = true,
+}: PageProps) => {
+  return (
+    <tr
+      className={cn({
+        ['contentUp']: true,
+        [`${tableStyles.link}`]: onClick,
+      })}
+      style={{ animationDelay: 55 + 3.5 * index + 'ms' }}
```

Comment:
do we always want the animation delay of all tables to be 55 ? 🐷

Reply:
Ignoring this comment 😭

Reply:
Im not sure.... I at least think it should be the same for all the tables in Veita..... right?

Reply:
you win 🏆